features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/RecordEnricher.java

```diff
@@ -66,42 +67,49 @@ public RecordEnricher(DnsResolver dnsResolver) {
         this.dnsResolver = Objects.requireNonNull(dnsResolver);
     }
 
-    public CompletableFuture<RecordEnrichment> enrich(Iterable<Value<?>> record) {
-        final IpAddressCapturingVisitor ipAddressCapturingVisitor = new IpAddressCapturingVisitor();
-        for (final Value<?> value : record) {
-            value.visit(ipAddressCapturingVisitor);
-        }
-        final Set<InetAddress> addressesToReverseLookup = ipAddressCapturingVisitor.getAddresses();
-        final Map<InetAddress, String> hostnamesByAddress = new HashMap<>(addressesToReverseLookup.size());
-        final CompletableFuture reverseLookupFutures[] = addressesToReverseLookup.stream()
-                .map(addr -> {
-                    LOG.trace("Issuing reverse lookup for: {}", addr);
-                    return dnsResolver.reverseLookup(addr).whenComplete((hostname, ex) -> {
-                        if (ex == null) {
-                            LOG.trace("Got reverse lookup answer for '{}': {}", addr, hostname);
-                            synchronized (hostnamesByAddress) {
-                                hostnamesByAddress.put(addr, hostname.orElse(null));
+    public CompletableFuture<RecordEnrichment> enrich(Iterable<Value<?>> record, boolean lookupsEnabled) {
```

Comment:
Given that `lookupsEnabled` won't change from one enrichment to another, I think we should move this into the constructor instead. One less argument to pass on each call.

Reply:
I think I did what you meant. Please review

Reply:
@j-white Do these changes look valid based on what you suggested?

Reply:
👍